revisions-information.md

```diff
@@ -0,0 +1,87 @@
+
+## Where can I see revisions?
+
+Packages with revisions will tell you so on their cabal page. You can always access the revisions of a package by appending `/revisions` to its url, like so: http://hackage.haskell.org/package/lens-3.5.1/revisions/
+
+## How can I download a package without revisions applied?
+
+`cabal get --pristine`
+
+## How can I operate without taking into account revisions?
+
+If you produced something at a particular point in time and want to pin it down so that no future revisions can alter anything about it, you can specify the `index-state` parameter (either in a configuration file or on the command line) as documented here: http://cabal.readthedocs.io/en/latest/nix-local-build.html?highlight=index-state#cfg-field-index-state
+
+This pins down the view of the package universe to use revisions as of any particular timestamp and so allows you to `cabal freeze` not only versions, but the exact revisions of them.
+
+## How are revisions numbered?
+
+Revisions are numbered by natural numbers, starting at zero, and increase in increments of 1.
+
+## What can revisions change?
+
+Revisions are intended to only update metadata used by build tooling, or that is used purely for display purposes. The source code of a package is never changed, only specified fields of cabal files. Given the same fixed dependencies and flag configuration, as long as a metadata change hasn't disallowed them, the build product from a compiler in the same environment is intended to be the same regardless of revision changes. Put another way, it is intended revisions do not alter the semantic meaning of the version assigned to a package. 
+
+Specifically, revisions can alter the following:
+
+* Dependency bounds (tightening or loosening)
+* Cabal spec version (only between the interval 1.10 and 1.21)
+* Custom-setup dependencies (adding, removing or altering bounds)
+* Flag defaults
+* Flag type (automatic to manual only)
+* Flag descriptions
+* Dependencies (adding, but only in very special cases, see below).
+* Copyright attribution
+* Maintainer attribution
+* Author attribution
+* Tested-with information
+* Bug report URL
+* Homepage URL
+* Source repositories
+* Package synopsis
+* Package description
+* Package category
+
+Revisions can also add custom-setup stanzas. This is necessary because newer cabal library versions make explicit custom-setup depends, while older versions simply shared exposed to `Setup.hs` whatever happened to be registered in the user pkg db, which was fragile and created a situation with untracked dependencies (cf: https://www.well-typed.com/blog/2015/07/cabal-setup-deps/)
```

Comment:
What about "while older versions simply shared exposed to `Setup.hs`" -> "In older versions  `Setup.hs` would have access to"?